### PR TITLE
fix(systemaddon): Add empty state support for sections. Closes #2568.

### LINKED ISSUE #2568: User friendly message is not displayed for empty sections
**[Affected versions]:**
- Firefox 55.0a1 & 53.0.2
- Activity Stream Build: 1.11.0-dev-1494274961-dc23841

**[Affected Platforms]:**
- All Windows
- All Mac
- All Linux

**[Prerequisites]:**
- Latest Activity Stream dev add-on version was previously installed on a profile with history.
- "Trending Stories" are activated from New Tab Preferences.

**[Steps to reproduce]:**
1. Start the browser with the profile from prerequisites.
2. Open a new tab and dismiss all the "Trending Stories" tiles.
3. Observe the "Trending Stories" section.
 
**[Expected results]:**
- An user friendly message is displayed. The user is informed that if he will return later there will be more Trending Stories. 

**[Actual results]:**
- After all "Trending Stories" tiles are removed, the section is hidden.

### PR DESCRIPTION
**Pull request by AdamHillier**

### CODE REVIEW

File: system-addon/content-src/components/Sections/_Sections.scss [hunk 1]
```diff
@@ -75,4 +77,43 @@
     margin: 0;
   }
 
+  .section-empty-state {
+    width: 100%;
+    height: 266px;
```
Comment (k88hudson) [inline system-addon/content-src/components/Sections/_Sections.scss]: Can you refactor `$card-height` to be available to both this file and _Card.scss so you can use it here?

File: system-addon/content-src/components/Sections/Sections.jsx [hunk 2]
```diff
@@ -33,6 +34,15 @@ class Section extends React.Component {
         {initialized && (<ul className="section-list" style={{padding: 0}}>
           {rows.map((link, index) => link && <Card index={index} dispatch={dispatch} link={link} />)}
         </ul>)}
+        {!initialized &&
+          <div className="section-empty-state">
+            <div className="empty-state">
+              <img className="empty-state-icon icon-top-stories" />
```
Comment (sarracini) [inline system-addon/content-src/components/Sections/Sections.jsx]: can this be generalized like:
```html
<img className={`empty-state-icon icon icon-${emptyState.icon}`} />
```

File: system-addon/content-src/components/Sections/_Sections.scss [hunk 3]
```diff
@@ -75,4 +77,43 @@
...
+        background-repeat: no-repeat;
+        background-position: center;
+        fill: $section-empty-grey;
+        -moz-context-properties: fill;
+        height: 50px;
+        width: 50px;
+        margin: 0 auto;
+        display: block;
+
+        &.icon-top-stories {
```
Comment (sarracini) [inline system-addon/content-src/components/Sections/_Sections.scss]: can you move this to _icons.scss:
```css
&.icon-check {
     background-image: url('#{$image-path}glyph-check-16.svg');
}
```

**Comment by as-pine-proxy** [comment]
[Treeherder: 2de387c375651d8cb694d95ab107557db48aea03](https://treeherder.mozilla.org/#/jobs?repo=pine&revision=2de387c375651d8cb694d95ab107557db48aea03)

**Pull request merged**